src/ui/pages/ConnectedSites/ConnectedSites.tsx

```diff
@@ -143,38 +149,131 @@ function ConnectedSitesList({
   );
 }
 
+function ConnectedSitesSearch({
+  value,
+  onChange,
+  inputRef,
+}: {
+  value?: string;
+  onChange(value: string): void;
+  inputRef: React.MutableRefObject<InputHandle | null>;
+}) {
+  return (
+    <DebouncedInput
+      ref={inputRef}
+      value={value || ''}
+      delay={300}
+      onChange={onChange}
+      render={({ value, handleChange }) => (
+        <SearchInput
+          boxHeight={40}
+          type="search"
+          placeholder="Search"
+          value={value}
+          onChange={(event) => {
+            handleChange(event.currentTarget.value);
+          }}
+        />
+      )}
+    />
+  );
+}
+
+function EmptyView({
+  hasConnectedSites,
+  hasFilters,
+  onReset,
+}: {
+  hasConnectedSites: boolean;
+  hasFilters: boolean;
+  onReset(): void;
+}) {
+  return (
+    <FillView>
+      {hasConnectedSites ? (
+        <VStack gap={6} style={{ textAlign: 'center' }}>
+          <UIText kind="headline/hero">🥺</UIText>
+          <UIText kind="small/accent" color="var(--neutral-500)">
+            <VStack gap={4}>
+              <div>No connected DApps found</div>
+              {hasFilters ? (
+                <UnstyledButton
+                  onClick={onReset}
+                  style={{ color: 'var(--primary)' }}
+                  className={helperStyles.hoverUnderline}
+                >
+                  Reset all filters
+                </UnstyledButton>
+              ) : null}
+            </VStack>
+          </UIText>
+        </VStack>
+      ) : (
+        <UIText
+          kind="body/regular"
+          color="var(--neutral-500)"
+          style={{ padding: 20, textAlign: 'center' }}
+        >
+          You will see a list of connected DApps here
+        </UIText>
+      )}
+    </FillView>
+  );
+}
+
 function ConnectedSitesMain() {
-  const { data: connectedSites, ...connectedSitesQuery } = useQuery({
+  const {
+    data: allConnectedSites,
+    isLoading,
+    ...connectedSitesQuery
+  } = useQuery({
     queryKey: ['getPermissionsWithWallets'],
     queryFn: getPermissionsWithWallets,
     useErrorBoundary: true,
     suspense: true,
   });
+  const [searchQuery, setSearchQuery] = useState<string | undefined>();
```

Comment:
```suggestion
  const [searchQuery, setSearchQuery] = useState<string | null>(null);
```
But also why not an empty string?